wolfssl/wolfcrypt/sp_int.h

```diff
@@ -741,22 +741,32 @@ typedef struct sp_ecc_ctx {
 /** Result of comparison is that the first number is less than second. */
 #define MP_LT    (-1)
 
+#ifdef WOLFSSL_DEBUG_TRACE_ERROR_CODES_H
+#include <wolfssl/debug-untrace-error-codes.h>
+#endif
+
 /* ERROR VALUES */
-/** Error value on success. */
-#define MP_OKAY          0
-/** Error value when dynamic memory allocation fails. */
-#define MP_MEM          (-2)
-/** Error value when value passed is not able to be used. */
-#define MP_VAL          (-3)
-/** Error value when non-blocking operation is returning after partial
- * completion.
- */
-#define FP_WOULDBLOCK   (-4)
+enum {
```

Comment:
Don't do this.
Revert the enum.
Why did you do it?

Reply:
I explained in my commit and PR messages:

>sp_int.h: provide for `WOLFSSL_DEBUG_TRACE_ERROR_CODES`, and refactor MP error codes as `enum`s, for consistency with other error codes.

>note, refactoring the sp error codes as `enum`s adds helpful additional error checking by the compiler, and also helped `cppcheck` uncover the benign-but-true `identicalInnerConditions`.


again, longstanding `master` code intermixes MP codes with `error-crypt.h` codes.  it was never consistent for one errcode namespace to be represented using macros while the other is represented with `enum`s.  and it was never correct for them to be intermixed, because they have been maintained in separate and uncoordinated namespaces.

I'll go ahead and update this PR with a commit that does solution #1, and we'll see what we think.

Reply:
PR now updated with the global fix.  worked out well.